lib/XMLHttpRequest.js

```diff
@@ -170,7 +173,8 @@ exports.XMLHttpRequest = function () {
       url: url.toString(),
       async: (typeof async !== 'boolean' ? true : async),
       user: user || null,
-      password: password || null
+      password: password || null,
+      rejectUnauthorized: (typeof rejectUnauthorized !== 'boolean' ? true : rejectUnauthorized)
```

Comment:
introduce rejectUnauthorized setting with default of true to enable easy https testing without a valid certificate.

Reply:
It's not part of [`XMLHttpRequest.open`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open). I'm not sure if it's possible to do an insecure request with `XMLHttpRequest`...

Reply:
Hmmm.... this class already diverges in capabilities from XMLHttpRequest, and this is an optional last parameter defaulting to secure.
Do you have an alternate way to test https? If so I'm happy to try it out.

Reply:
> Hmmm.... this class already diverges in capabilities from XMLHttpRequest, 

Could you expand on that? The original goal of this library was to provide a compatible implementation of `XMLHttpRequest` in Node so we should try to stick to it.

> Do you have an alternate way to test https? If so I'm happy to try it out.

I guess we will need to somehow override/mock the settings in the test without altering the API (production code).
Maybe Sinon.js could help.

Reply:
It doesn't support most of the return types. What's the problem with supporting a non-standard extension that doesn't interfere with anything if you don't use it?
I didn't look hard, but it does't look to me as if sinon.js actually uses https requests.
If you insist on removing the open parameter, I'll see if there's some way to directly set something on the instance to accomplish the same thing.

Reply:
> I didn't look hard, but it does't look to me as if sinon.js actually uses https requests.

Sinon.js can create stubs and mocks, so you could declare a `getHttpOptions` function and stub it in the test to pass the `rejectUnauthorized`.

> It doesn't support most of the return types. 

If some use cases are not supported I think it's fair but we should try to adhere to the specification when we implement something.

> What's the problem with supporting a non-standard extension that doesn't interfere with anything if you don't use it?

It does interfere since the function now has 6 parameters. Arguably, it's not a big deal but I don't feel like it's the right way to do.
Doing an insecure request is simply not available/possible in `XMLHttpRequest`.

As you said, we only need it in the tests suite (to do an HTTPS request without a valid certificate) so we should not alter the production code.

Reply:
Apparently we can avoid the parameter by setting
```
process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = 0;
```
in the appropriate tests.